store/cachekv/internal/btree_test.go

```diff
@@ -0,0 +1,203 @@
+	verifyIterator(t, itr, []int64{7}, "forward iterator from 7 to 8")
+
+	ritr, err = db.ReverseIterator(int642Bytes(4), int642Bytes(5))
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64{4}, "reverse iterator from 5 (ex) to 4")
+
+	ritr, err = db.ReverseIterator(int642Bytes(4), int642Bytes(6))
+	require.NoError(t, err)
+	verifyIterator(t, ritr,
+		[]int64{5, 4}, "reverse iterator from 6 (ex) to 4")
+
+	ritr, err = db.ReverseIterator(int642Bytes(4), int642Bytes(7))
+	require.NoError(t, err)
+	verifyIterator(t, ritr,
+		[]int64{5, 4}, "reverse iterator from 7 (ex) to 4")
+
+	ritr, err = db.ReverseIterator(int642Bytes(5), int642Bytes(6))
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64{5}, "reverse iterator from 6 (ex) to 5")
+
+	ritr, err = db.ReverseIterator(int642Bytes(5), int642Bytes(7))
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64{5}, "reverse iterator from 7 (ex) to 5")
+
+	ritr, err = db.ReverseIterator(int642Bytes(6), int642Bytes(7))
+	require.NoError(t, err)
+	verifyIterator(t, ritr,
+		[]int64(nil), "reverse iterator from 7 (ex) to 6")
+
+	ritr, err = db.ReverseIterator(int642Bytes(10), nil)
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64(nil), "reverse iterator to 10")
+
+	ritr, err = db.ReverseIterator(int642Bytes(6), nil)
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64{9, 8, 7}, "reverse iterator to 6")
+
+	ritr, err = db.ReverseIterator(int642Bytes(5), nil)
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64{9, 8, 7, 5}, "reverse iterator to 5")
+
+	ritr, err = db.ReverseIterator(int642Bytes(8), int642Bytes(9))
+	require.NoError(t, err)
+	verifyIterator(t, ritr, []int64{8}, "reverse iterator from 9 (ex) to 8")
+
+	ritr, err = db.ReverseIterator(int642Bytes(2), int642Bytes(4))
+	require.NoError(t, err)
+	verifyIterator(t, ritr,
+		[]int64{3, 2}, "reverse iterator from 4 (ex) to 2")
+
+	ritr, err = db.ReverseIterator(int642Bytes(4), int642Bytes(2))
+	require.NoError(t, err)
+	verifyIterator(t, ritr,
+		[]int64(nil), "reverse iterator from 2 (ex) to 4")
+
+	// Ensure that the iterators don't panic with an empty database.
+	db2 := NewBTree()
+
+	itr, err = db2.Iterator(nil, nil)
+	require.NoError(t, err)
+	verifyIterator(t, itr, nil, "forward iterator with empty db")
+
+	ritr, err = db2.ReverseIterator(nil, nil)
+	require.NoError(t, err)
+	verifyIterator(t, ritr, nil, "reverse iterator with empty db")
+}
+
+func verifyIterator(t *testing.T, itr *memIterator, expected []int64, msg string) {
+	i := 0
+	for itr.Valid() {
+		key := itr.Key()
+		require.Equal(t, expected[i], bytes2Int64(key), "iterator: %d mismatches", i)
+		itr.Next()
+		i++
+	}
+	require.Equal(t, i, len(expected), "expected to have fully iterated over all the elements in iter")
+}
+
+func int642Bytes(i int64) []byte {
+	buf := make([]byte, 8)
+	binary.BigEndian.PutUint64(buf, uint64(i))
+	return buf
+}
+
+func bytes2Int64(buf []byte) int64 {
+	return int64(binary.BigEndian.Uint64(buf))
+}
```

Comment:
Pretty sure we have public methods in `types` for this already. I'd do a quick search as to not keep duplicating code.

Reply:
done, updated.